tests/aws/services/transcribe/test_transcribe.py

```diff
@@ -1,28 +1,120 @@
 import logging
 import os
+import threading
+import time
 from urllib.parse import urlparse
 
 import pytest
 from botocore.exceptions import ClientError
 
 from localstack.aws.api.transcribe import BadRequestException, ConflictException, NotFoundException
 from localstack.aws.connect import ServiceLevelClientFactory
+from localstack.packages.ffmpeg import ffmpeg_package
+from localstack.services.transcribe.packages import vosk_package
+from localstack.services.transcribe.provider import LANGUAGE_MODELS, TranscribeProvider
 from localstack.testing.pytest import markers
 from localstack.utils.files import new_tmp_file
 from localstack.utils.strings import short_uid, to_str
 from localstack.utils.sync import poll_condition, retry
+from localstack.utils.threads import start_worker_thread
 
 BASEDIR = os.path.abspath(os.path.dirname(__file__))
 
 LOG = logging.getLogger(__name__)
 
+# Lock and event to ensure that the installation is executed before the tests
+INIT_VOSK_LOCK = threading.Lock()
+INIT_FFMPEG_LOCK = threading.Lock()
+vosk_installed = threading.Event()
+ffmpeg_installed = threading.Event()
+installation_errored = threading.Event()
+
+INSTALLATION_TIMEOUT = 5 * 60
+PRE_DOWNLOAD_LANGUAGE_CODE_MODELS = ["en-GB"]
+
+
+def install_async():
+    """
+    Installs the default ffmpeg and vosk versions in a worker thread.
+    """
+    if vosk_installed.is_set() and ffmpeg_installed.is_set():
+        return
+
+    def install_vosk(*args):
+        with INIT_VOSK_LOCK:
+            if vosk_installed.is_set():
+                return
+            try:
+                LOG.info("installing Vosk default version")
+                vosk_package.install()
+                LOG.info("done installing Vosk default version")
+                LOG.info(
+                    "downloading Vosk models used in test: %s", PRE_DOWNLOAD_LANGUAGE_CODE_MODELS
+                )
+                for language_code in PRE_DOWNLOAD_LANGUAGE_CODE_MODELS:
+                    model_name = LANGUAGE_MODELS[language_code]
+                    # downloading the model takes quite a while sometimes
+                    TranscribeProvider.download_model(model_name)
+                    LOG.info(
+                        "done downloading Vosk model '%s' for language code '%s'",
+                        model_name,
+                        language_code,
+                    )
+                LOG.info("done downloading all Vosk models used in test")
+            except Exception:
+                LOG.exception("Error during installation of Vosk dependencies")
+                installation_errored.set()
+                # we also set the other event to quickly stop the polling
+                ffmpeg_installed.set()
+            finally:
+                vosk_installed.set()
```

Comment:
For another iteration:
This is actually an interesting use case, which could be interesting for other package installers as well.
If you have a service with multiple dependencies, we currently do not really execute them in parallel.
We could implement a native `async_install` and `await_install` functions in the `PackageInstaller` / `Package` abstraction, which would allow users of multiple package installers to easily implement a fork-join pattern to trigger them in parallel...

But again, something for another time 😛

Reply:
Definitely, it's something I thought about as well, we could use a thread pool executor to dispatch all the `async_install` too, and wait on all the futures returned. Interesting ideas!